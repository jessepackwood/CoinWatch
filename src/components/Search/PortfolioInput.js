import React from 'react'

const PortfolioInput = () => {
  return (
    <div>
      <input 
        className='portfolio-add-coin' 
        type='number' label='Number of coins' 
        placeholder='Amount' value={amount || ''} 
        onChange={(event)=> handleAmountInputChange(event.target)}
      />
    </div>
  )
}

export default PortfolioInput