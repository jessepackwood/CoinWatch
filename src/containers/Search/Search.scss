.search-bar {
	font-size: 20pt;
	height: 40px;
	width: 95%;
	margin-top: 20px;
	padding: 5px;
	border: 3px solid #274156;
	border-radius: 6px;
	outline: none;
}

.search-wrapper {
	display: flex;
	justify-content: center;
}