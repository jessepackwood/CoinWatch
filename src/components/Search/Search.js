import React from 'react'
import './Search.css'

export const Search = () => {
	return (
		<div className='search-wrapper'>
			<input className='search-bar' type='text' placeholder='Search' />
		</div>
		)
}

export default Search