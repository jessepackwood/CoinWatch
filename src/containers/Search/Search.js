import React from 'react'
import { connect } from 'react-redux'
import * as actions from '../../actions'
import PropTypes from 'prop-types'
import './Search.css'

export const Search = ({ inputChange }) => {

  return (
    <div className='search-wrapper'>
      <input 
        className='search-bar' 
        type='text' 
        placeholder='Search by coin name' 
        onChange={event => inputChange(event.target.value)} 
      />
    </div>
  )
}

export const mapStateToProps = state => ({
  searchInput: state.searchInput.value
})

export const mapDispatchToProps = dispatch => {
  return {
    inputChange: (searchInput) => {
      dispatch(actions.searchInputChange(searchInput))
    }
  }
}

Search.propTypes = {
  inputChange: PropTypes.func
}

export default connect(mapStateToProps, mapDispatchToProps)(Search)