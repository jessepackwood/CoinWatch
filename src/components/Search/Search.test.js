import React from 'react';
import { shallow } from 'enzyme';
import { Search, mapStateToProps, mapDispatchToProps } from './Search';

describe('Search tests', () => {
  let mockCoins
  let mockSearchInput
  let renderedSearch
  let inputChange

  beforeEach(() => {
    mockCoins = [{long:'Bitcoin'}]
    mockSearchInput = 'bit'
    inputChange = jest.fn()
    renderedSearch = shallow(
      <Search searchInput={mockSearchInput} coins={mockCoins} inputchange={inputChange}/>
    );
  });

  it('Should match the snapshot', () => {
    expect(renderedSearch).toMatchSnapshot();
  });

  describe('mapStateToProps tests', () => {
    it('Should pull coins from the store', () => {
      const mockSearchInput = {value: 'bit'}
      const mockStore = {
        coins: mockCoins,
        searchInput: mockSearchInput.value
      };
      const result = mapStateToProps(mockStore);

      expect(result.coins).toEqual(mockStore.coins);
    });
  });

  describe('mapDispatchToProps tests', () => {
    it('Should call dispatch when inputChange is called', () => {
      const mockDispatch = jest.fn();
      const result = mapDispatchToProps(mockDispatch);

      result.inputChange('bit');
      expect(mockDispatch).toHaveBeenCalled();
    })
  })
})