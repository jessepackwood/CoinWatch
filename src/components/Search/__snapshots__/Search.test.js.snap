// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    coins={
        Array [
            Object {
              "long": "Bitcoin",
            },
          ]
    }
    inputchange={[Function]}
    searchInput="bit"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="search-bar"
        onChange={[Function]}
        placeholder="Search by coin name"
        type="text"
/>,
      "className": "search-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "search-bar",
        "onChange": [Function],
        "placeholder": "Search by coin name",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="search-bar"
          onChange={[Function]}
          placeholder="Search by coin name"
          type="text"
/>,
        "className": "search-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-bar",
          "onChange": [Function],
          "placeholder": "Search by coin name",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
